packages/@react-aria/tag/test/useTagGroup.test.js

```diff
@@ -0,0 +1,113 @@
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {fireEvent, render, within} from '@react-spectrum/test-utils';
+import {Item} from '@react-stately/collections';
+import React from 'react';
+import {useButton} from '@react-aria/button';
+import {useListState} from '@react-stately/list';
+import userEvent from '@testing-library/user-event';
+import {useTag, useTagGroup} from '../';
+
+function TagGroup(props) {
+  let ref = React.useRef(null);
+  let state = useListState(props);
+  let {gridProps} = useTagGroup(props, state, ref);
+
+  return (
+    <div ref={ref} className="tag-group">
+      <div {...gridProps}>
+        {[...state.collection].map((item) => (
+          <Tag
+            key={item.key}
+            item={item}
+            state={state}>
+            {item.rendered}
+          </Tag>
+        ))}
+      </div>
+    </div>
+  );
+}
+
+function Tag(props) {
+  let {item, state} = props;
+  let ref = React.useRef(null);
+  let {rowProps, gridCellProps, removeButtonProps, allowsRemoving} = useTag(props, state, ref);
+
+  return (
+    <div ref={ref} {...rowProps}>
+      <div {...gridCellProps}>
+        {item.rendered}
+        {allowsRemoving && <Button {...removeButtonProps}>x</Button>}
+      </div>
+    </div>
+  );
+}
+
+function Button(props) {
+  let ref = React.useRef(null);
+  let {buttonProps} = useButton(props, ref);
+  return <button {...buttonProps} ref={ref}>{props.children}</button>;
+}
+
+describe('useTagGroup', function () {
+  it('should support selection', () => {
+    let onRemove = jest.fn();
+    let {getAllByRole} = render(
+      <TagGroup
+        label="Amenities"
+        selectionMode="multiple"
+        defaultSelectedKeys={['parking']}
+        onRemove={onRemove}>
+        <Item key="laundry">Laundry</Item>
+        <Item key="fitness">Fitness center</Item>
+        <Item key="parking">Parking</Item>
+        <Item key="pool">Swimming pool</Item>
+        <Item key="breakfast">Breakfast</Item>
+      </TagGroup>
+    );
+
+    let tags = getAllByRole('row');
+    expect(tags).toHaveLength(5);
+    expect(tags[0]).toHaveAttribute('aria-selected', 'false');
+    expect(tags[1]).toHaveAttribute('aria-selected', 'false');
+    expect(tags[2]).toHaveAttribute('aria-selected', 'true');
+    expect(tags[3]).toHaveAttribute('aria-selected', 'false');
+    expect(tags[4]).toHaveAttribute('aria-selected', 'false');
+
+    userEvent.click(tags[3]);
+    expect(tags[0]).toHaveAttribute('aria-selected', 'false');
+    expect(tags[1]).toHaveAttribute('aria-selected', 'false');
+    expect(tags[2]).toHaveAttribute('aria-selected', 'true');
+    expect(tags[3]).toHaveAttribute('aria-selected', 'true');
+    expect(tags[4]).toHaveAttribute('aria-selected', 'false');
+
+    fireEvent.keyDown(tags[3], {key: 'Backspace'});
```

Comment:
I assume clicking on the remove button when two are selected would also remove both.

Reply:
That's what we do below, on line 107.

Reply:
Right, I missed that onRemove is doing nothing so the selection or the list size is never changing.